src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistHistoryScreen.vue

```diff
@@ -0,0 +1,218 @@
+              />
+            </oxd-grid-item>
+          </oxd-grid>
+        </oxd-form-row>
+        <oxd-form-row>
+          <oxd-grid :cols="3" class="orangehrm-full-width-grid">
+            <oxd-grid-item
+              class="orangehrm-save-candidate-page --span-column-2"
+            >
+              <oxd-input-field
+                :label="$t('general.notes')"
+                type="textarea"
+                :v-model="history.note"
+              />
+            </oxd-grid-item>
+          </oxd-grid>
+        </oxd-form-row>
+        <oxd-divider />
+        <oxd-form-actions>
+          <oxd-button
+            display-type="ghost"
+            :label="$t('general.back')"
+            @click="onBack"
+          />
+          <submit-button :label="$t('general.save')" />
+        </oxd-form-actions>
+      </oxd-form>
+    </div>
+  </div>
+</template>
+
+<script>
+import {APIService} from '@/core/util/services/api.service';
+import {navigate} from '@/core/util/helper/navigation';
+import StatusInput from '@/orangehrmRecruitmentPlugin/components/StatusInput';
+export default {
+  name: 'ShortlistHistoryScreen',
+  components: {'status-input': StatusInput},
+  setup() {
+    const http = new APIService(
+      'https://0d188518-fc5f-4b13-833d-5cd0e9fcef79.mock.pstmn.io',
+      'recruitment/candidateHistory',
+    );
+    return {
+      http,
+    };
+  },
+  data() {
+    return {
+      isLoading: false,
+      history: {
+        candidate: '',
+        vacancy: '',
+        manager: '',
+        performedAction: '',
+        performedBy: '',
+        performedDate: '',
+        note: '',
+        statusId: '',
+      },
+      actions: [
+        {
+          id: 1,
+          label: 'Shortlist',
+        },
+        {
+          id: 2,
+          label: 'Schedule Interview',
+        },
+        {
+          id: 3,
+          label: 'Mark Interview',
+        },
+        {
+          id: 4,
+          label: 'Passed',
+        },
+        {
+          id: 5,
+          label: 'Failed',
+        },
+        {
+          id: 6,
+          label: 'Offer Job',
+        },
+        {
+          id: 7,
+          label: 'Decline Offer',
+        },
+        {
+          id: 7,
+          label: 'Hire',
+        },
+        {
+          id: 8,
+          label: 'Reject',
+        },
+      ],
+    };
+  },
+  beforeMount() {
+    this.isLoading = true;
+    this.http.getAll().then(({data: {data}}) => {
+      const {candidate, performedAction, vacancy, manager, ...rest} = data;
+      const {firstName, lastName, middleName} = candidate;
+      const fullName = `${manager.firstName} ${manager.middleName} ${manager.lastName}`;
+      this.history = {
+        candidate: `${firstName} ${middleName} ${lastName}`,
+        vacancy: vacancy.title,
+        manager:
+          (manager?.terminationId ? this.$t('general.past_employee') : '') +
+          fullName,
+        performedAction: this.actions.find(({id}) => id === performedAction)
```

Comment:
action will come as object with id and label from API recruitment/candidateHistory'. no need to have a array and pick from array.